probe-rs/src/debug/unit_info.rs

```diff
@@ -441,6 +419,57 @@ impl UnitInfo {
         Ok(())
     }
 
+    /// Extract the array range values
+    fn extract_array_range_attribute(
+        &self,
+        entry: &gimli::DebuggingInformationEntry<GimliReader>,
+    ) -> Result<Option<std::ops::Range<u64>>, DebugError> {
+        let mut variable_attributes = entry.attrs();
+
+        let mut lower_bound = None;
+        let mut upper_bound = None;
+
+        // Now loop through all the unit attributes to extract the remainder of the `Variable` definition.
+        while let Ok(Some(attr)) = variable_attributes.next() {
+            match attr.name() {
+                // Property of variables that are of DW_TAG_subrange_type.
+                gimli::DW_AT_lower_bound => match attr.value().udata_value() {
+                    Some(bound) => lower_bound = Some(bound),
+                    None => {
+                        return Err(DebugError::Other(anyhow::anyhow!(
+                            "Unimplemented: Attribute Value for DW_AT_lower_bound: {:?}",
+                            attr.value()
+                        )));
+                    }
+                },
+                // Property of variables that are of DW_TAG_subrange_type.
+                gimli::DW_AT_upper_bound | gimli::DW_AT_count => {
+                    match attr.value().udata_value() {
+                        Some(bound) => upper_bound = Some(bound), // child_variable.range_upper_bound = bound as i64,
+                        None => {
+                            return Err(DebugError::Other(anyhow::anyhow!(
+                                "Unimplemented: Attribute Value for DW_AT_upper_bound: {:?}",
+                                attr.value()
+                            )));
+                        }
+                    }
+                }
+                other_attribute => {
+                    tracing::info!(
```

Comment:
Won't this be overly verbose at `info` level?

Reply:
It shouldn't happen, but I can reduce it.

Reply:
Not sure what you mean by "shouldn't happen" but I have elfs with subrange types that have `DW_AT_type` for example and I don't see why those wouldn't be logged here.